core/test/transform_sparse_and_dense_test.cpp

```diff
@@ -0,0 +1,159 @@
+  std::vector<span<float>> spans_val{weights_.values<float>()};
+  std::vector<span<float>> spans_var{weights_.variances<float>()};
+  auto weights_view = makeVariable<span<float>>({}, spans_val, spans_var);
+
+  DataArray hist(weights_, {{Dim::X, edges_}});
+
+  const auto out = transform<std::tuple<
+      std::tuple<sparse_container<double>, span<double>, span<float>>>>(
+      var, edges_view, weights_view,
+      overloaded{
+          [](const auto &sparse, const auto &edges, const auto &weights) {
+            expect::histogram::sorted_edges(edges);
+            using W = std::decay_t<decltype(weights)>;
+            constexpr bool have_variance =
+                core::detail::is_ValueAndVariance_v<W>;
+            using T = sparse_container<
+                typename core::detail::element_type_t<W>::value_type>;
+            T out_vals;
+            T out_vars;
+            out_vals.reserve(sparse.size());
+            if (have_variance)
+              out_vars.reserve(sparse.size());
+            if (scipp::numeric::is_linspace(edges)) {
+              auto len = scipp::size(sparse) - 1;
+              const auto offset = edges.front();
+              const auto nbin = static_cast<decltype(offset)>(len);
+              const auto scale = nbin / (edges.back() - edges.front());
+              for (const auto c : sparse) {
+                const auto bin = (c - offset) * scale;
+                if (bin >= 0.0 && bin < nbin) {
+                  if constexpr (have_variance) {
+                    out_vals.emplace_back(weights.value[bin]);
+                    out_vars.emplace_back(weights.variance[bin]);
+                  } else {
+                    out_vals.emplace_back(weights[bin]);
+                  }
+                } else {
+                  // TODO should we set NAN instead?
+                  out_vals.emplace_back(1.0);
+                  if constexpr (have_variance)
+                    out_vars.emplace_back(1.0);
+                }
+              }
+            } else {
+              throw std::runtime_error("Only equal-sized bins supported.");
+            }
+            if constexpr (have_variance) {
+              return std::pair(std::move(out_vals), std::move(out_vars));
+            } else {
+              return out_vals;
+            }
+          },
+          transform_flags::expect_no_variance_arg<0>,
+          transform_flags::expect_no_variance_arg<1>,
+          [](const units::Unit &a, const units::Unit &b, const units::Unit &c) {
+            expect::equals(a, b);
+            return c;
+          }});
+  EXPECT_TRUE(out.hasVariances());
+  const auto out_vals = out.sparseValues<float>();
+  EXPECT_TRUE(equals(out_vals[0], {2, 3, 3}));
+  EXPECT_TRUE(equals(out_vals[1], {2, 2, 3, 1}));
+  const auto out_vars = out.sparseVariances<float>();
+  EXPECT_TRUE(equals(out_vars[0], {0.3f, 0.4f, 0.4f}));
+  EXPECT_TRUE(equals(out_vars[1], {0.3f, 0.3f, 0.4f, 1.0f}));
+}
```

Comment:
Do we need `sparse_time_dense` as a function in API? In other case I would say this test is  to sophisticated and complex.

Also no tests for incorrect arguments and corner cases.

Reply:
Sorry, I should have clarified this more in the PR description: Essentially this is a prototype for what I will implement for #676 in the next step. The code in these tests will be split into two parts:
1. Helper to generate the `Variable` with `T=span`.
2. Operator that does the multiplication (and division).

This test will likely just disappear, or be replaced by something more meaningful. For now it was mainly to check that we can compile what needs to be compiled.